File: sakila-payment-service/src/main/java/com/example/sakila/module/rental/repository/RentalRepository.java
package com.example.sakila.module.rental.repository;

import com.example.sakila.module.rental.Rental;

import java.util.List;

public interface RentalRepository {

  Rental getRentalById(Long id);

  List<Rental> getRentalsByInventoryId(Long id);

  List<Rental> getRentalsByCustomerId(Long id);

  List<Rental> getRentalsByStaffId(Long id);

}

File: sakila-payment-service/src/main/java/com/example/sakila/module/rental/repository/RentalRepository.kt
package com.example.sakila.module.rental.repository

import com.example.sakila.module.rental.Rental

interface RentalRepository {

  fun getRentalById(id: Long): Rental?

  fun getRentalsByCustomerId(id: Long): List<Rental>?

  fun getRentalsByStaffId(id: Long): List<Rental>?

  fun insertRental(rental: Rental): Rental

  fun updateRental(rental: Rental): Rental

  fun deleteRental(rental: Rental)

}
